### ISSUE TITLE
Using R: api error: Training data only contains one class

### ISSUE THREAD

**Issue by mangstad** [Microsoft/LightGBM#252]
I installed LightGBM in R on Linux (Ubuntu 12.04). Everything installed fine.

When trying to run a simple test on my data, I encountered the following error:
`
Error in lgb.call("LGBM_DatasetCreateFromMat_R", ret = handle, private$raw_data,  : 
  api error: Training data only contains one class
`

Tracked it down to happening when explicitly constructing the lgb.Dataset object (the error doesn't happen at the time of lgb.Dataset call. Only happens when trying to run training, or explicitly calling lgb.Dataset.construct).

A simple reproducible example is:
```
library(lightgbm)
sample.data = matrix(rnorm(1000),nrow=100,ncol=10)
sample.labels = round(runif(100))
sample.train = lgb.Dataset(sample.data,label=sample.labels)
nround=2
param=list(num_leaves=4,learning_rate=1,objective="binary")
lgb.cv(param,sample.train,nround,nfold=5,eval={"binary_error"})
```

The sample.labels definitely do not contain only one value:
```
> summary(sample.labels)
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   0.00    0.00    0.00    0.41    1.00    1.00 
```

Actually, I just double checked and I get the same behavior from just following the cross_validation.R demo using the provided dataset.

**Comment by guolinke** [comment]
@mangstad 
weird. I can run your code successfully...
Did you try to build with latest code?

**Comment by mangstad** [comment]
I installed from github 20-30 minutes before submitting that issue, so it was the latest then. I'll try again today.

**Comment by guolinke** [comment]
@mangstad How do you install it ?

**Comment by mangstad** [comment]
As suggested in the readme in the R package folder
`devtools::install_github("Microsoft/LightGBM", subdir = "R-package")
`

**Comment by guolinke** [comment]
maybe it is caused by ```Ubuntu 12.04```. I don't have environment to test. 
can you run it successfully in other platforms?

**Comment by yanyachen** [comment]
I had same problem when I was using Ubuntu 14.04.
You can solve it by restarting your R session, at least that works for me.
But I can not reproduce that error with latest code for now.

**Comment by mangstad** [comment]
Restarting the session didn't do anything. However, I just reinstalled from github, and apparently one of the commits in the past couple days must have fixed this as I no longer get the error.
